src/main/java/com/laserfiche/api/client/APIServer/TokenClientImpl.java

```diff
@@ -0,0 +1,120 @@
+                .build();
+    }
+
+    @Override
+    public CompletableFuture<SessionKeyInfo> createAccessToken(String repoId, CreateConnectionRequest body) {
+        Map<String, Object> pathParameters = getNonNullParameters(new String[]{"repoId"},
+                new Object[]{repoId});
+        CompletableFuture<HttpResponse<SessionKeyInfo>> future = Unirest
+                .post(baseUrl + "/v1/Repositories/{repoId}/Token")
+                .routeParam(pathParameters)
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/x-www-form-urlencoded")
+                .field("grant_type", "password")
+                .field("username", body
+                        .getUsername()
+                        .toString())
+                .field("password", body
+                        .getPassword()
+                        .toString())
+                .asObjectAsync(SessionKeyInfo.class);
+        return future.thenApply(httpResponse -> {
+            if (httpResponse.getStatus() != 200) {
+                ProblemDetails problemDetails;
+                try {
+                    String jsonString = new JSONObject(httpResponse.getBody()).toString();
+                    problemDetails = objectMapper.readValue(jsonString, ProblemDetails.class);
+                } catch (JsonProcessingException e) {
+                    e.printStackTrace();
+                    return null;
+                }
+                Map<String, String> headersMap = getHeadersMap(httpResponse);
+                if (httpResponse.getStatus() == 400)
+                    throw new ApiException("Invalid or bad request.", httpResponse.getStatus(),
+                            httpResponse.getStatusText(), headersMap, problemDetails);
+                else if (httpResponse.getStatus() == 401)
+                    throw new ApiException("Access token is invalid or expired.", httpResponse.getStatus(),
+                            httpResponse.getStatusText(), headersMap, problemDetails);
+                else if (httpResponse.getStatus() == 403)
+                    throw new ApiException("Access denied for the operation.", httpResponse.getStatus(),
+                            httpResponse.getStatusText(), headersMap, problemDetails);
+                else if (httpResponse.getStatus() == 404)
+                    throw new ApiException("Requested attribute key not found.", httpResponse.getStatus(),
+                            httpResponse.getStatusText(), headersMap, problemDetails);
+                else if (httpResponse.getStatus() == 429)
+                    throw new ApiException("Rate limit is reached.", httpResponse.getStatus(),
+                            httpResponse.getStatusText(), headersMap, problemDetails);
+                else
+                    throw new RuntimeException(httpResponse.getStatusText());
+            }
+            return httpResponse.getBody();
```

Comment:
We will never reach this line?

Reply:
hmm strange how this one file isn't updated after i pushed the latest changes, but this line should no longer be here after my latest commit

Reply:
I think Ray is working on something so all cloud dev is not working right now. I am working on getting a new environment on cloud test but it's not done.